In [9]:
import requests
import lxml
from bs4 import BeautifulSoup
from xlwt import *

#Table Initialization
paperlist = Workbook(encoding = 'utf-8')
table = paperlist.add_sheet('data')
table.write(0, 0, 'Title')
table.write(0, 1, 'year')
table.write(0, 2, 'abstract')
line = 1

#Scrape paper
keyword = "한국운동역학회"

for start in range(0,91,10):
    print(start)
    url = "https://scholar.google.co.kr/scholar?start="+str(start)+"&q="+keyword+"&hl=ko&as_sdt=0,5"
    print(url)
    headers = {
      'User-Agent': 'Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/63.0.3239.132 Safari/537.36 QIHU 360SE'
    }
    f = requests.get(url)
    soup = BeautifulSoup(f.content, 'lxml')
    paperTags = soup.find_all("div",{
       'class': 'gs_rt gs_a gs_rs gs_ri'
     })
    print(f)
    for anchor in paperTags:
        print("paper")
        paper = anchor.find_next("div", {
            'class': 'gs_ri'
        })
        title = paper.find_next("h3").get_text()
        year = paper.find_next("div",{'class': 'gs_a'}).get_text()
        abstract = paper.find_next("div").get_text()
    
        print(title)
        print(year)
        print(abstract)
        table.write(line, 0, title)
        table.write(line, 1, year)
        table.write(line, 2, abstract)
        line += 1
        print(paperlist)

paperlist.save('한국운동역학회1_10.xlsx')

0
https://scholar.google.co.kr/scholar?start=0&q=한국운동역학회&hl=ko&as_sdt=0,5
<Response [429]>
10
https://scholar.google.co.kr/scholar?start=10&q=한국운동역학회&hl=ko&as_sdt=0,5
<Response [429]>
20
https://scholar.google.co.kr/scholar?start=20&q=한국운동역학회&hl=ko&as_sdt=0,5
<Response [429]>
30
https://scholar.google.co.kr/scholar?start=30&q=한국운동역학회&hl=ko&as_sdt=0,5
<Response [429]>
40
https://scholar.google.co.kr/scholar?start=40&q=한국운동역학회&hl=ko&as_sdt=0,5
<Response [429]>
50
https://scholar.google.co.kr/scholar?start=50&q=한국운동역학회&hl=ko&as_sdt=0,5
<Response [429]>
60
https://scholar.google.co.kr/scholar?start=60&q=한국운동역학회&hl=ko&as_sdt=0,5
<Response [429]>
70
https://scholar.google.co.kr/scholar?start=70&q=한국운동역학회&hl=ko&as_sdt=0,5
<Response [429]>
80
https://scholar.google.co.kr/scholar?start=80&q=한국운동역학회&hl=ko&as_sdt=0,5
<Response [429]>
90
https://scholar.google.co.kr/scholar?start=90&q=한국운동역학회&hl=ko&as_sdt=0,5
<Response [429]>


* 100페이지 까지 했을때 response 429로 너무 많은 요청 보냄으로 거절
* 혹시 모르니 50페이지씩 2로 나눠 크롤링 예정 -> 이었으나 아직 거절인 상태 추후 10개의 페이지로 나눠서 크롤링 예정